conda/recipe/0001-c++14.patch

```diff
@@ -0,0 +1,25 @@
+From 286b45047a4bb7036896d27c1201955ed3692593 Mon Sep 17 00:00:00 2001
+From: "Andres A. Lugo-Reyes" <aalugore@us.ibm.com>
+Date: Mon, 22 Jul 2019 21:02:50 +0000
+Subject: [PATCH] set std=c++14
+
+---
+ CMakeLists.txt | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 7e07808..31f6206 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -97,7 +97,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
+ set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -std=c++14")
+ 
+ # CXX flags
+-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-unused-variable -Wno-unused-function -fno-strict-aliasing -fPIC -fvisibility=hidden")
++set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -Wall -Wno-unused-variable -Wno-unused-function -fno-strict-aliasing -fPIC -fvisibility=hidden")
```

Comment:
Few lines above we are setting `std=c++14` using cmake
```
# Use -std=c++14 (and not gnu++14)
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```
Could you provide more info on why it was necessary to add `-std=c++14` to `CMAKE_CXX_FLAGS` as well?

Reply:
This was added to keep in sync with the nvcc flags. Without it we get a bunch of errors like so:
ppc:
```
/root/anaconda3/conda-bld/dali_1565722501972/_build_env/powerpc64le-conda_cos7-linux-gnu/include/c++/8.2.0/bits/stl_p
air.h(79): error: inline specifier allowed on function declarations only                                             
                                                                                                                     
/root/anaconda3/conda-bld/dali_1565722501972/_build_env/powerpc64le-conda_cos7-linux-gnu/include/c++/8.2.0/bits/stl_p
air.h(437): error: argument list for class template "std::pair" is missing                                           
                                                                                                                     
/root/anaconda3/conda-bld/dali_1565722501972/_build_env/powerpc64le-conda_cos7-linux-gnu/include/c++/8.2.0/bits/stl_p
air.h(437): error: expected a ")"                                                                                    
                                                                                                                     
/root/anaconda3/conda-bld/dali_1565722501972/_build_env/powerpc64le-conda_cos7-linux-gnu/include/c++/8.2.0/bits/stl_p
air.h(437): error: template parameter "_T1" may not be redeclared in this scope                                      
                                                                                                                     
/root/anaconda3/conda-bld/dali_1565722501972/_build_env/powerpc64le-conda_cos7-linux-gnu/include/c++/8.2.0/bits/stl_p
air.h(437): error: expected a ";"                                                                                    
                                                                                                                     
/root/anaconda3/conda-bld/dali_1565722501972/_build_env/powerpc64le-conda_cos7-linux-gnu/include/c++/8.2.0/utility(3$
3): error: inline specifier allowed on function declarations only                                                    
                                                                                                                     
/root/anaconda3/conda-bld/dali_1565722501972/_build_env/powerpc64le-conda_cos7-linux-gnu/include/c++/8.2.0/bits/stl_v
ector.h(1737): error: argument list for class template "std::vector" is missing                                      
                                                                                                                     
/root/anaconda3/conda-bld/dali_1565722501972/_build_env/powerpc64le-conda_cos7-linux-gnu/include/c++/8.2.0/bits/stl_v
ector.h(1737): error: expected a ")"   
```
and x86:
```
/opt/anaconda3/conda-bld/dali_1565721773519/_build_env/x86_64-conda_cos6-linux-gnu/include/c++/7.2.0/bits/stl_pair.h(79)
: error: inline specifier allowed on function declarations only

/opt/anaconda3/conda-bld/dali_1565721773519/_build_env/x86_64-conda_cos6-linux-gnu/include/c++/7.2.0/bits/stl_pair.h(437
): error: argument list for class template "std::pair" is missing

/opt/anaconda3/conda-bld/dali_1565721773519/_build_env/x86_64-conda_cos6-linux-gnu/include/c++/7.2.0/bits/stl_pair.h(437
): error: expected a ")"

/opt/anaconda3/conda-bld/dali_1565721773519/_build_env/x86_64-conda_cos6-linux-gnu/include/c++/7.2.0/bits/stl_pair.h(437
): error: template parameter "_T1" may not be redeclared in this scope

/opt/anaconda3/conda-bld/dali_1565721773519/_build_env/x86_64-conda_cos6-linux-gnu/include/c++/7.2.0/bits/stl_pair.h(437
): error: expected a ";"

/opt/anaconda3/conda-bld/dali_1565721773519/_build_env/x86_64-conda_cos6-linux-gnu/include/c++/7.2.0/utility(366): error
: inline specifier allowed on function declarations only

/opt/anaconda3/conda-bld/dali_1565721773519/_build_env/x86_64-conda_cos6-linux-gnu/include/c++/7.2.0/array(244): error:
argument list for class template "std::array" is missing

/opt/anaconda3/conda-bld/dali_1565721773519/_build_env/x86_64-conda_cos6-linux-gnu/include/c++/7.2.0/array(244): error:
expected a ")"
```